jetstream/fakespot-in-suggest.toml

```diff
@@ -0,0 +1,203 @@
+[metrics]
+weekly = ["admarketplace_sponsored_impressions",
+          "fakespot_amazon_impressions",
+          "fakespot_bestbuy_impressions",
+          "fakespot_walmart_impressions",
+          "fakespot_amazon_impression_rate",
+          "fakespot_bestbuy_impression_rate",
+          "fakespot_walmart_impression_rate",
+          "fakespot_amazon_clicks",
+          "fakespot_bestbuy_clicks",
+          "fakespot_walmart_clicks",
+          "fakespot_amazon_ctr",
+          "fakespot_bestbuy_ctr",
+          "fakespot_walmart_ctr",
+          "fakespot_amazon_not_relevant_dismissals",
+          "fakespot_bestbuy_not_relevant_dismissals",
+          "fakespot_walmart_not_relevant_dismissals",
+          "fx_urlbar_quick_suggest_remote_settings_latency_ms"]
+
+overall = ["admarketplace_sponsored_impressions",
+           "fakespot_amazon_impressions",
+           "fakespot_bestbuy_impressions",
+           "fakespot_walmart_impressions",
+           "fakespot_amazon_impression_rate",
+           "fakespot_bestbuy_impression_rate",
+           "fakespot_walmart_impression_rate",
+           "fakespot_amazon_clicks",
+           "fakespot_bestbuy_clicks",
+           "fakespot_walmart_clicks",
+           "fakespot_amazon_ctr",
+           "fakespot_bestbuy_ctr",
+           "fakespot_walmart_ctr",
+           "fakespot_amazon_not_relevant_dismissals",
+           "fakespot_bestbuy_not_relevant_dismissals",
+           "fakespot_walmart_not_relevant_dismissals",
+           "fx_urlbar_quick_suggest_remote_settings_latency_ms"]
+
+## Impressions
+[metrics.admarketplace_sponsored_impressions]
+select_expression = "COUNT(DISTINCT IF(is_terminal AND result.result_type like '%adm_sponsored', event_id, NULL))"
+data_source = "urlbar_events_unnested_results"
+[metrics.admarketplace_sponsored_impressions.statistics.bootstrap_mean]
+
+[metrics.fakespot_amazon_impressions]
+select_expression = "COUNT(DISTINCT IF(is_terminal AND result.result_type = 'rust_fakespot_amazon', event_id, NULL))"
+data_source = "urlbar_events_unnested_results"
+[metrics.fakespot_amazon_impressions.statistics.bootstrap_mean]
+
+[metrics.fakespot_bestbuy_impressions]
+select_expression = "COUNT(DISTINCT IF(is_terminal AND result.result_type = 'rust_fakespot_bestbuy', event_id, NULL))"
+data_source = "urlbar_events_unnested_results"
+[metrics.fakespot_bestbuy_impressions.statistics.bootstrap_mean]
+
+[metrics.fakespot_walmart_impressions]
+select_expression = "COUNT(DISTINCT IF(is_terminal AND result.result_type = 'rust_fakespot_walmart', event_id, NULL))"
+data_source = "urlbar_events_unnested_results"
+[metrics.fakespot_walmart_impressions.statistics.bootstrap_mean]
+
+[metrics.fakespot_impressions]
+select_expression = "COUNT(DISTINCT IF(is_terminal AND result.result_type in ('rust_fakespot_amazon', 'rust_fakespot_bestbuy', 'rust_fakespot_walmart'), event_id, NULL))"
```

Comment:
same here. I'd like to keep the business logic in that UDF if possible, so it's not scattered across multiple repos. https://github.com/mozilla/bigquery-etl/blob/main/sql/mozfun/norm/result_type_to_product_name/udf.sql

Reply:
since this is a new product type for the experiment/pilot, we haven't brought into the udf mentioned. I think that would be useful thing to do if we decide to move forward with the feature but as of now, this would be the best way to calculate the impression